Library/Homebrew/download_strategy.rb

```diff
@@ -459,7 +459,7 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
     return @resolved_info_cache[url] if @resolved_info_cache.include?(url)
 
     if (domain = Homebrew::EnvConfig.artifact_domain)
-      url = url.sub(%r{^(https?://#{GitHubPackages::URL_DOMAIN}/)?}o, "#{domain.chomp("/")}/")
+      url = url.sub(%r{^https?://#{GitHubPackages::URL_DOMAIN}/}o, "#{domain.chomp("/")}/")
```

Comment:
It seems like this may still be a desirable change? Thoughts?

Reply:
not really if i understand the codebase correctly. while this logic is implemented in a Curl base class, which is used by a variety of ther download stategies, it really only affects `GitHubPackages::URL_DOMAIN`, which is where the GithubPackages download stategy comes into play; and that can be influence with the `HOMEBREW_BOTTLE_DOMAIN` variable.

going forward with this change would essentially make `HOMEBREW_BOTTLE_DOMAIN` and `HOMEBREW_ARTIFACT_DOMAIN` somewhat equal (albeight being implemented differently)

with #13313 on the horizon, even authentication can be
supported for both scenarios.

Reply:
Yeah I had [my original PR](https://github.com/Homebrew/brew/pull/13288), but it was missing the tests so I will get my local env cleaned up and see about reopening my original one with this fix, the doc update, and the tests.